src/base/io/utilities/convertOldCouplingFormat.m

```diff
@@ -0,0 +1,135 @@
+function model = convertOldCouplingFormat(model)
+%Converts an old style model implementation of coupling constraints to a
+%new style
+%
+% INPUT
+% model     model with model.A but without model.d   
+%
+% OUTPUT
+%    model:     A COBRA model structure with the following fields
+%
+%                * `.S` - The stoichiometric matrix
+%                * `.c` - Objective coeff vector
+%                * `.lb` - Lower bound vector
+%                * `.ub` - Upper bound vector              
+%                  * `.b`: accumulation/depletion vector (default 0 for each metabolite).
+%                  * `.C`: the Constraint matrix;
+%                  * `.d`: the right hand side vector for C;
+%                  * `.dsense`: the constraint sense vector;
+
+if isfield(model,'A') && isfield(model,'S')
+%     if printLevel >=1
+%         warning('The inserted Model contains an old style coupling matrix (A). The MAtrix will be converted into a Coupling Matrix (C) and fields will be adapted.')
+%     end
+    [nMets,nRxns] = size(model.S);
+    if size(model.S,1) == size(model.A,1)
+        bool=strncmp('slack_',model.mets,length('slack_'));
+        if any(bool)
+            %this is a draft harvey or harvetta
```

Comment:
Can you provide me with a model of this type? 
Because at the moment, this code breaks a few things and I would like to correct it in a way that works for both these draft models and other models.